### PR TITLE
tests: mailbox: Added new test cases.

### PR DESCRIPTION
**Pull request by AdithyaBaglody**
Added new test cases to increase the branch coverage.

### CODE REVIEW

File: tests/kernel/mbox/mbox_api/testcase.yaml [hunk 1]
```diff
@@ -1,3 +1,4 @@
 tests:
   kernel.mailbox:
     tags: kernel
+    platform_exclude: quark_d2000_crb nucleo_f030r8
```
Comment (SavinayDharmappa) [inline tests/kernel/mbox/mbox_api/testcase.yaml]: add newline

Comment (nashif) [reply tests/kernel/mbox/mbox_api/testcase.yaml]: is that because of ram? in this case please add min_ram: 16 instead of listing platforms

Comment (AdithyaBaglody) [reply tests/kernel/mbox/mbox_api/testcase.yaml]: fixed it.

**Comment by ramakrishnapallala** [comment]
commit message can be more detailed or informative.
Have consistency in test naming like mbox.

**Comment by AdithyaBaglody** [comment]
@ramakrishnapallala fixed it .

File: tests/kernel/mbox/mbox_api/src/test_mbox_api.c [hunk 2]
```diff
@@ -60,6 +74,47 @@ static char data[MAX_INFO_TYPE][MAIL_LEN] = {
...
+		break;
+
+	case 4:
+		mmsg.rx_source_thread = K_ANY;
+		break;
+
+	default:
+		break;
+	}
+		mmsg.size = 0;
```
Comment (andrewboie) [inline tests/kernel/mbox/mbox_api/src/test_mbox_api.c]: please fix indentation

File: tests/kernel/mbox/mbox_api/src/test_mbox_api.c [hunk 3]
```diff
@@ -185,19 +368,181 @@ static void tmbox_get(struct k_mbox *pmbox)
 		zassert_true(k_mbox_get(pmbox, &mmsg, NULL, K_FOREVER) == 0,
 			     NULL);
 		zassert_true(k_mbox_data_block_get
-				     (&mmsg, &mpoolrx, &rxblock, K_FOREVER) == 0, NULL);
```
Comment (andrewboie) [inline tests/kernel/mbox/mbox_api/src/test_mbox_api.c]: why change this line?

File: tests/kernel/mbox/mbox_api/src/test_mbox_api.c [hunk 4]
```diff
@@ -185,19 +368,181 @@ static void tmbox_get(struct k_mbox *pmbox)
...
+		mmsg.rx_source_thread = random_tid;
+		zassert_true(k_mbox_get(pmbox, &mmsg, NULL, TIMEOUT) == -EAGAIN,
+			     NULL);
+		break;
+	case BLOCK_GET_INVALID_POOL:
+		/* To dispose of the rx msg using block get */
+		mmsg.rx_source_thread = K_ANY;
+		k_mbox_get(pmbox, &mmsg, NULL, K_FOREVER);
+		zassert_true(k_mbox_data_block_get(&mmsg,
+						   NULL,
```
Comment (andrewboie) [inline tests/kernel/mbox/mbox_api/src/test_mbox_api.c]: why are the args split out one per line like this?

**Review by tyrantlizard77 — CHANGES_REQUESTED** [review]
logic seems to be alright, but some style issues...

File: tests/kernel/mbox/mbox_api/src/test_mbox_api.c (hunk 3 above)
Comment (AdithyaBaglody) [reply tests/kernel/mbox/mbox_api/src/test_mbox_api.c]: Was the only line in the file which exceeded beyond 80 character. Hence fixed it.

**Comment by AdithyaBaglody** [comment]
Delta:
Handled the review comments.

**Review by ramakrishnapallala — APPROVED** [review]

**Comment by AdithyaBaglody** [comment]
@andrewboie can we merge this PR?

**Review by tyrantlizard77 — APPROVED** [review]

**Pull request merged**